test: TestHandleCertificateChange in gateway_test.go fails intermittently

<!--

This issue tracker is a best-effort forum for users and customers to report bugs.

Be sure to not include any sensitive information. Sensitive information should __NOT__ be included in this issue.

-->

**Bug description**:
The unit test is flaky and fails intermittently in the CI (likely a timing issue in the test):
```
=== RUN   TestHandleCertificateChange/setting_spec_when_not_previously_set
2:19PM TRC Releasing certificate foo.bar.cluster.local component=tresor file=certificate_manager.go:155
    gateway_test.go:367: 
        	Error Trace:	gateway_test.go:367
        	Error:      	Condition never satisfied
        	Test:       	TestHandleCertificateChange/setting_spec_when_not_previously_set
        	Messages:   	Expected secret was not found
2:19PM INF Proxy update dispatcher received stop signal, exiting component=message-broker file=broker.go:189
--- FAIL: TestHandleCertificateChange (4.36s)
    --- FAIL: TestHandleCertificateChange/setting_spec_when_not_previously_set (4.36s)
```

**Affected area (please mark with X where applicable)**:
- Tests                      [X]

--- a/pkg/ingress/gateway_test.go
+++ b/pkg/ingress/gateway_test.go
@@ -24,6 +24,7 @@ import (
 )
 
 const (
+	maxTimeToSubscribe           = 500 * time.Millisecond
 	maxTimeForEventToBePublished = 2 * time.Second
 	maxSecretPollTime            = 2 * time.Second
 	secretPollInterval           = 25 * time.Millisecond
@@ -335,7 +336,7 @@ func TestHandleCertificateChange(t *testing.T) {
 
 			go c.handleCertificateChange(tc.previousCertSpec, tc.stopChan)
 			defer close(tc.stopChan)
-			time.Sleep(maxTimeForEventToBePublished)
+			time.Sleep(maxTimeToSubscribe)
 
 			// If a secret is supposed to exist, create it
 			if tc.previousCertSpec != nil {
@@ -349,6 +350,7 @@ func TestHandleCertificateChange(t *testing.T) {
 					NewObj: tc.updatedMeshConfig,
 					OldObj: tc.previousMeshConfig,
 				}, announcements.MeshConfigUpdated.String())
+				time.Sleep(maxTimeForEventToBePublished)
 			}
 
 			if !tc.expectSecretToExist {

